### PR TITLE
Move `Strict` to `Function.Strict` and move `_$!_` there

### PR DESCRIPTION
**Pull request by MatthewDaggitt**
In response to #1486. Haven't yet cut the dependency of `Function.Base` on `Function.Strict` in order to preserve backwards compatibility.

### CODE REVIEW

**Comment by MatthewDaggitt** [comment]
Question - should `Function` re-export `Function.Strict`?

**Comment by MatthewDaggitt** [comment]
> Question - should Function re-export Function.Strict?

In the absence of any comment, I've gone with "yes".

File: src/Strict.agda [hunk 1]
```diff
@@ -1,31 +1,20 @@
 ------------------------------------------------------------------------
 -- The Agda standard library
 --
--- Strictness combinators
+-- This module is DEPRECATED, please use `Data.Record` directly.
```
Comment (JacquesCarette) [inline src/Strict.agda:4]: `Data.Record` ?

**Review by JacquesCarette — CHANGES_REQUESTED** [review]
Other than a minor typo, this is great.

File: src/Strict.agda (hunk 1 above)
Comment (MatthewDaggitt) [reply src/Strict.agda:4]: Thanks, good spot!

**Pull request merged**